packages/@lightningjs/ui-components/src/mixins/withThemeStyles/utils.js

```diff
@@ -274,173 +268,400 @@ const findPropertiesBySubProperty = (obj, subPropertyName) => {
+  }
+
+  if (!Array.isArray(subProps)) {
+    throw new TypeError('Expected subProps to be an array of strings.');
+  }
+
+  return [
+    ...new Set([...defaultProps, ...Object.keys(additionalProps), ...subProps])
+  ];
+}
+
+/**
+ * Generate the payload by cloning and merging multiple objects.
+ *
+ * @param {Object} base - The base object to start with.
+ * @param {Object} defaultStyle - Default styles provided by the user.
+ * @param {string} toneItem - The current tone being processed.
+ * @param {string} modeItem - The current mode being processed.
+ * @param {Object} tone - Tone configurations.
+ * @param {Object} mode - Mode configurations.
+ * @returns {Object} - The merged payload.
+ */
+export function generatePayload(
+  base,
+  defaultStyle,
+  toneItem,
+  modeItem,
+  tone,
+  mode
+) {
+  let payload = clone(defaultStyle, base);
+  payload = clone(payload, tone?.[toneItem]);
+  payload = clone(payload, mode?.[modeItem]);
+  payload = clone(payload, tone?.[toneItem]?.mode?.[modeItem] || {});
+  payload = clone(payload, mode?.[modeItem]?.tone?.[toneItem] || {});
+  return payload;
+}
+
+/**
+ * Generates a solution based on the provided configurations.
+ *
+ * @param {Object} options - The configuration options for generating the solution.
+ * @param {Object} [options.base={}] - Base object.
+ * @param {Object} [options.tone={}] - Tone configurations.
+ * @param {Object} [options.mode={}] - Mode configurations.
+ * @param {Object} [options.defaultStyle={}] - Default styles provided by the user.
+ * @returns {Object} - The generated solution with shared references and merged identical properties.
+ */
+export const generateSolution = ({
+  base = {},
+  tone = {},
+  mode = {},
+  defaultStyle = {}
+}) => {
+  const solution = {};
+
+  const toneProperties = findPropertiesBySubProperty(mode, 'tone');
+  const modeProperties = findPropertiesBySubProperty(tone, 'mode');
+
+  const uniqueModes = getUniqueProperties(
+    ['unfocused', 'focused', 'disabled'],
+    mode,
+    modeProperties
+  );
+  const uniqueTones = getUniqueProperties(
+    ['neutral', 'inverse', 'brand'],
+    tone,
+    toneProperties
+  );
+
+  for (const modeItem of uniqueModes) {
+    for (const toneItem of uniqueTones) {
+      const payload = generatePayload(
+        base,
+        defaultStyle,
+        toneItem,
+        modeItem,
+        tone,
+        mode
+      );
+      solution[`${modeItem}_${toneItem}`] = payload;
+    }
+  }
+
+  return solution;
+};
+
+const DEFAULT_KEYS = [
+  'unfocused_neutral',
+  'unfocused_inverse',
+  'unfocused_brand',
+  'focused_neutral',
+  'focused_inverse',
+  'focused_brand',
+  'disabled_neutral',
+  'disabled_inverse',
+  'disabled_brand'
+];
+
+const FALLBACK_ORDER = [
+  'unfocused_neutral',
+  'unfocused_inverse',
+  'unfocused_brand',
+  'focused_neutral',
+  'focused_inverse',
+  'focused_brand',
+  'disabled_neutral',
+  'disabled_inverse',
+  'disabled_brand'
```

Comment:
Since these are the same as above, do we really need both variables?

Reply:
Yes you are correct. We don't need both